Source/Eto/Forms/TrayIndicator.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using Eto.Drawing;
+
+namespace Eto.Forms
+{
+    /// <summary>
+    /// A system tray indicator.
+    /// </summary>
+    [Handler(typeof(TrayIndicator.IHandler))]
+    public class TrayIndicator : Widget
+    {
+        private Icon icon;
+
+        new IHandler Handler { get { return (IHandler)base.Handler; } }
+
+        static readonly object callback = new Callback();
+
+        /// <summary>
+        /// Gets an instance of an object used to perform callbacks to the widget from handler implementations.
+        /// </summary>
+        /// <returns>The callback instance to use for this widget.</returns>
+        protected override object GetCallback() { return callback; }
+
+        /// <summary>
+        /// Gets or sets the icon for the <see cref="TrayIndicator"/>.
+        /// </summary>
+        /// <value>The icon of the <see cref="TrayIndicator"/>.</value>
+        public Icon Icon
+        {
+            get { return icon; }
+            set
+            {
+                icon = value;
+                Handler.SetIcon(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the title/tooltip for the <see cref="TrayIndicator"/>.
+        /// </summary>
+        /// <value>The title/tooltip for the <see cref="TrayIndicator"/>.</value>
+        public string Title
+        {
+            get { return Handler.Title; }
+            set { Handler.Title = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this <see cref="TrayIndicator"/> is visible.
+        /// </summary>
+        /// <value><c>true</c> if visible; otherwise, <c>false</c>.</value>
+        public bool Visible
+        {
+            get { return Handler.Visible; }
+            set { Handler.Visible = value; }
+        }
+
+        /// <summary>
+        /// Hide this instance of <see cref="TrayIndicator"/>.
+        /// </summary>
+        /// <remarks>
+        /// Make sure to call this method before closing the application.
+        /// </remarks>
+        public void Hide()
+        {
+            Visible = false;
+        }
+
+        /// <summary>
+        /// Sets the indicator menu.
+        /// </summary>
+        /// <remarks>
+        /// Make sure to call this event every time you make changes to the context menu.
+        /// </remarks>
+        /// <param name="menu">The indicator menu.</param>
+        public void SetMenu(ContextMenu menu)
```

Comment:
Could we change this to a property instead?  Could be done similarly to how you've done the Icon property.  It will also prevent the ContextMenu from being GC'd if the handler doesn't keep a reference to it.